source/agora/consensus/EnrollmentManager.d

```diff
@@ -255,45 +255,97 @@ public class EnrollmentManager
-            enroll = will contain the Enrollment if created
+            utxo = The hash of the frozen UTXO used as a stake.
+                   It must be owned by the private key this validator controls
+                   (`key_pair` argument to constructor).
 
         Returns:
-            true if the enrollment manager succeeded in creating the Enrollment
+            The `Enrollment` created by `buildEnrollment`
 
     ***************************************************************************/
 
-    public bool createEnrollment (Hash frozen_utxo_hash, Height height,
-        out Enrollment enroll) @trusted nothrow
+    public Enrollment createEnrollment (Hash utxo) @safe nothrow
     {
         // K, frozen UTXO hash
-        this.data.utxo_key = frozen_utxo_hash;
-        this.enroll_key = frozen_utxo_hash;
-
-        // N, cycle length
-        this.data.cycle_length = this.params.ValidatorCycle;
+        this.enroll_key = utxo;
 
         // X, final seed data and preimages of hashes
         //
         // TODO: Move this consume call to `addValidator` and instead
         // make this calculate the future pre-image without changing
         // the state
-        this.data.random_seed = this.cycle.consume(this.key_pair.v);
+        const seed = this.cycle.consume(this.key_pair.v);
+        this.data = makeEnrollment(
+            this.key_pair, utxo, this.params.ValidatorCycle,
+            seed, this.cycle.index);
 
-        // R, signature noise
-        this.signature_noise = this.createSignatureNoise(height);
+        return this.data;
+    }
 
-        // signature
-        data.enroll_sig = sign(this.key_pair, this.signature_noise, this.data);
+    /***************************************************************************
 
-        enroll = this.data;
+        Build enrollment data for an arbitrary utxo + key combination
 
-        return true;
+        This static function builds a valid, real-world 'enrollment' for any
+        private key + utxo combination.  It can be used to generate realistic
+        test data or specialized usage, e.g. by tools.
+
+        To avoid code duplication and overhead, this function has two overload:
+        the one accepting an offset will generate a new `PreImageCache`,
+        which is quite expensive, and retrieve the hash to use based on the
+        offset (defaults to `0`), and then forwards to the second overload
+        that accepts a `Hash` and thus does not incur the cost of re-generating
+        the pre-images.
+
+        Params:
+            key  = `KeyPair` that controls the `utxo`
+            utxo = `Hash` of a frozen UTXO value that can be used for enrollment
+            cycle_length = The cycle length to use (see `ConsensusParams`)
+            offset = The number of times this private key has enrolled before.
+                     If `seed` is provided, this parameter is non-optional.
+            seed = Random seed to use
```

Comment:
A good reading is done if the sort order of the parameter descriptions is the same as the parameter order of the function.

Reply:
What do you mean ?